includes/frontend/blocks/class-crowdsignal-forms-nps-block.php

```diff
@@ -168,6 +168,14 @@ private function attributes() {
 				'type'    => 'string',
 				'default' => 3,
 			),
+			'status'              => array(
```

Comment:
Nit: Just an observation I made... in terms of the block itself, it works just as well if we replace status with an `isOpen` or `isClosed` property. Then we could simplify the conditions to:

```
if isClosed:
    return;

// If a survey isn't closed but closedAfter is set, it implies the 'close after' state
if closedAfter && closedAfter < currentDate:
    return;

// display block
```

The advantage of this approach is it removes the need for the constants entirely which removes any overhead otherwise necessary to share the logic between the client and the server.

That said, if we're going to simplify it, we should probably do that before we launch the block as not to have to deal with supporting both ways for already saved blocks.

Reply:
Well, we could have a nice function to return `isClosed` value based on the status and date. Kinda looks the same to me as a `isClosed` attribute could only be derived via a function/calculation you mention, but open to suggestions. I always go for the *raw data* approach and then derive values depending on context.

Reply:
I am talking about simplifying the raw data though, by replacing the tri-state `status` with a binary `isClosed` value. This constrains the domain a bit which gets rid of some theoretical undefined states.

But like I said, it's nitpicking. I just can't resist the opportunity to write less code :D